src/OrchardCore.Modules/Orchard.Users/Services/UserStore.cs

```diff
@@ -286,33 +298,45 @@ public Task SetNormalizedEmailAsync(User user, string normalizedEmail, Cancellat
         #endregion
 
         #region IUserRoleStore<User>
-        public Task AddToRoleAsync(User user, string roleName, CancellationToken cancellationToken)
+        public async Task AddToRoleAsync(User user, string normalizedRoleName, CancellationToken cancellationToken = default(CancellationToken))
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
+            var roleNames = await _roleProvider.GetRoleNamesAsync();
+            var roleName = roleNames?.FirstOrDefault(r => _normalizeKey.Normalize(r) == normalizedRoleName);
+
+            if (string.IsNullOrWhiteSpace(roleName))
+            {
+                throw new InvalidOperationException(T["Role {0} does not exist.", normalizedRoleName]);
```

Comment:
I don't think exceptions should be translated, unless it's reused upper in the call chain to display in the UI, then please ignore.